### ISSUE TITLE
display title of theorems are in a xdef

### ISSUE THREAD

**Issue by u-fischer** [T-F-S/tcolorbox#165]
The following document fails 

~~~~
\documentclass{article}
\usepackage[theorems]{tcolorbox}
\newtcbtheorem{theorem}{Théoreme}{}{}
\begin{document}
\begin{theorem}{}{}
\end{theorem}
\end{document}
~~~~

with 

~~~~
\protect 
! Missing } inserted.
<inserted text> 
                }
~~~~

The problem imho is that the display title is x-expanded

~~~~
\__tcobox_theo_title:nnn #1#2#3->\tl_set:Nx \l_tmpa_tl {\__tcobox_theo_format:n
n {#1}{#2}}
....
#1<-ThÃ©oreme
#2<-\thetcbcounter 
#3<-
~~~~

**Comment by muzimuzhi** [comment]
The same constraint applies for the number part (the second-arg of `\__tcobox_theo_title:nnn`): `\the<counter>` is not necessarily fully expandable.

**Comment by T-F-S** [comment]
I will exchange the expansion with another code to check, if the theorem has an empty head.

**Comment by T-F-S** [comment]
### Fixed [5.0.2]
- Library `theorems`: New implementation had title expansion with problems (issue #165)
